community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/LuceneDocumentStructure.java

```diff
@@ -143,22 +143,22 @@ public static Query newRangeSeekByStringQuery( String lower, boolean includeLowe
     public static Query newWildCardStringQuery( String searchFor )
     {
         String searchTerm = QueryParser.escape( searchFor );
-        Term term = new Term( ValueEncoding.String.key(), "*" + searchTerm + "*" );
+        Term term = new Term( ValueEncoding.String.key( 0 ), "*" + searchTerm + "*" );
 
         return new WildcardQuery( term );
     }
 
     public static Query newRangeSeekByPrefixQuery( String prefix )
     {
-        Term term = new Term( ValueEncoding.String.key(), prefix );
+        Term term = new Term( ValueEncoding.String.key( 0 ), prefix );
         return USE_LUCENE_STANDARD_PREFIX_QUERY ? new PrefixQuery( term ) :
                                      new PrefixMultiTermsQuery( term );
     }
 
     public static Query newSuffixStringQuery( String suffix )
     {
         String searchTerm = QueryParser.escape( suffix );
-        Term term = new Term( ValueEncoding.String.key(), "*" + searchTerm );
+        Term term = new Term( ValueEncoding.String.key( 0 ), "*" + searchTerm );
```

Comment:
This still needs to be updated so we get the field position right, doesn't it?

Likewise for the `NonUniqueLucenePopulator` and `NonUniqueLucenePopulatingUpdater`, right?

Otherwise it looks like we are only populating our composite indexes with the first property.